cmd/triggerRun/cmd/root.go

```diff
@@ -157,31 +179,60 @@ func processTriggerSpec(client triggersclientset.Interface, tri *triggersv1.Trig
 		return nil, errors.New("Trigger is not defined")
 	}
 
+	//convert trigger to eventListener
 	el, err := triggersv1.ToEventListenerTrigger(tri.Spec)
 	if err != nil {
 		return nil, fmt.Errorf("fail to convert Trigger to EvenetListener: %w", err)
 	}
 
 	log := eventLog.With(zap.String(triggersv1.TriggerLabelKey, el.Name))
 
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		log.Fatalf("Failed to get in cluster config: %v", err)
+	}
+
+	kubeClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Fatalf("Failed to get the Kubernetes client set: %v", err)
+	}
+
+	r := sink.Sink{
+		KubeClientSet: kubeClient,
+		HTTPClient:    http.DefaultClient,
+		Auth:          sink.DefaultAuthOverride{},
+	}
+
+	finalPayload, header, err := r.ExecuteInterceptors(&el, request, event, log)
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+
 	rt, err := template.ResolveTrigger(el,
 		client.TriggersV1alpha1().TriggerBindings(tri.Namespace).Get,
 		client.TriggersV1alpha1().ClusterTriggerBindings().Get,
 		client.TriggersV1alpha1().TriggerTemplates(tri.Namespace).Get)
 	if err != nil {
-		log.Error(err)
+		log.Error("Fail to resolve Trigger", err)
 		return nil, err
 	}
 
-	params, err := template.ResolveParams(rt, event, request.Header)
+	params, err := template.ResolveParams(rt, finalPayload, header)
 	if err != nil {
-		log.Error(err)
+		log.Error("Fail to resolve parameters", err)
```

Comment:
Same input as above, this might be better as "Failed to resolve Trigger" or similar